File: backend/src/main/java/com/voltpay/backend/repository/UserRepository.java
package com.voltpay.backend.repository;

import com.voltpay.backend.model.User;
import org.springframework.data.mongodb.repository.MongoRepository;
import java.util.Optional;

public interface UserRepository extends MongoRepository<User, String> {
    Optional<User> findByPhoneNumber(String phoneNumber);
    Optional<User> findByUpiId(String upiId);
}

File: backend/src/main/kotlin/com/voltpay/backend/repository/UserRepository.kt
package com.voltpay.backend.repository

import com.voltpay.backend.model.User
import org.springframework.data.mongodb.repository.MongoRepository
import java.util.Optional

interface UserRepository : MongoRepository<User, String> {
    fun findByPhoneNumber(phoneNumber: String): Optional<User>
    fun findByUpiId(upiId: String): Optional<User>
}
